android/java/org/chromium/chrome/browser/app/BraveActivity.java

```diff
@@ -720,12 +725,44 @@ protected void initializeStartupMetrics() {
         BraveHelper.DisableFREDRP();
     }
 
+    @Override
+    public void onPreferenceChange() {
+        solveAdaptiveCaptcha();
+    }
+
+    public void solveAdaptiveCaptcha() {
+        String captchaID = UserPrefs.get(Profile.getLastUsedRegularProfile())
+                                   .getString(BravePref.SCHEDULED_CAPTCHA_ID);
+        String paymentID = UserPrefs.get(Profile.getLastUsedRegularProfile())
+                                   .getString(BravePref.SCHEDULED_CAPTCHA_PAYMENT_ID);
+        if (!TextUtils.isEmpty(captchaID) && !TextUtils.isEmpty(paymentID)
+                && !BravePrefServiceBridge.getInstance().getSafetynetCheckFailed()) {
+            BraveAdaptiveCaptchaUtils.solveCaptcha(captchaID, paymentID);
+        }
+    }
+
     @Override
     public void finishNativeInitialization() {
         super.finishNativeInitialization();
 
         BraveHelper.maybeMigrateSettings();
 
+        PrefChangeRegistrar mPrefChangeRegistrar = new PrefChangeRegistrar();
+        mPrefChangeRegistrar.addObserver(BravePref.SCHEDULED_CAPTCHA_ID, this);
```

Comment:
this observer is used when captchaid is assigned. so that we can try to solve the captcha right away. and on each app open, we also check for the captchaID